operational/stranded_servers/Stranded_Servers.pt

```diff
@@ -0,0 +1,191 @@
+  type "list"
+  label "Tags to ignore"
+  description "List of tags that will exclude Servers from being evaluated by this policy. Multiple tags are evaluated as an 'OR' condition. Tag must be of the format 'namespace:predicate=value'. Example: 'rs_agent:type=right_link_lite,rs_monitoring:state=auth'"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_rs", type: "rightscale"
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_list_stranded_instances" do
+  request do
+    run_script $js_list_stranded_instances, rs_cm_host, rs_project_id, $param_exclude_tags
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "cloud_name", jmes_path(col_item, "links.cloud.name")
+      field "cloud_type", jmes_path(col_item, "links.cloud.cloud_type")
+      field "name", jmes_path(col_item, "name")
+      field "resource_uid", jmes_path(col_item, "resource_uid")
+      field "updated_at", jmes_path(col_item, "links.deployment.timestamps.updated_at")
+      field "locked", jmes_path(col_item, "locked")
+      field "href", jmes_path(col_item, "href")
+      field "tags", jmes_path(col_item, "tags")
+      field "state", jmes_path(col_item, "state")
+    end
+  end
+end
+
+datasource "ds_get_stranded_instances" do
+  run_script $js_get_stranded_instance, $ds_list_stranded_instances, $param_exclude_tags
+end
+
+###############################################################################
+# Script
+###############################################################################
+
+script "js_list_stranded_instances", type: "javascript" do
+  result "request"
+  parameters "rs_cm_host","rs_project_id","param_exclude_tags"
+  code <<-EOS
+    var filtered_tags = ""
+    if ( param_exclude_tags.length > 0 ){
+      var filtered_tags = "&tag!=" + param_exclude_tags.join(",");
+    }
+
+    request = {
+      "auth": "auth_rs",
+      "verb": "GET",
+      "host": rs_cm_host,
+      "path": "/api/instances",
+      "headers": {
+        "X-Api-Version": "1.6",
+        "X-Account": \""+rs_project_id+\""
+      },
+      "query_params": {
+        "view": "full",
+        "filter": "state=stranded in booting" + filtered_tags
+      }
+    }
+
+  EOS
+end
+
+script "js_get_stranded_instance", type: "javascript" do
+  parameters "ds_list_stranded_instances","param_exclude_tags"
+  result "content"
+  code <<-EOS
+    var content=[]
+
+    for(var i=0;i<ds_list_stranded_instances.length;i++){
+      instance=ds_list_stranded_instances[i]
+	  
+	  // Constructing tags with comma separated to display in detail_template
+      var tags = instance['tags']
+      var dis_tags=""
+      for(var j=0;j<tags.length;j++){
```

Comment:
While this works just fine, it's more efficient to use join().  

`var dis_tags=tags.join(',) `

Reply:
@cdwilhelm 
I tried that using join() but the first two letters are getting missed and all tags are displayed as a link. So I iterated and did join, manually.

![image](https://user-images.githubusercontent.com/47381949/58631212-786a1400-82ff-11e9-9611-08a69d805ebc.png)

Reply:
that is very strange.  i've never seen that before